### PR TITLE
Make = and => optional in function declarations

### PR DESCRIPTION
**Pull request by nrc**
Also makes return types use `:` and formats parameter and return types

### CODE REVIEW

File: src/wasm-lib/kcl/src/parser/parser_impl.rs [hunk 1]
```diff
@@ -949,25 +949,31 @@ fn if_expr(i: TokenSlice) -> PResult<BoxNode<IfExpression>> {
 }
 
 // Looks like
-// (arg0, arg1) => {
+// (arg0, arg1) {
 //     const x = arg0 + arg1;
 //     return x
 // }
```
Comment (jtran) [inline src/wasm-lib/kcl/src/parser/parser_impl.rs:955]: Am I understanding it right that we allow function expressions that look like this, without `fn` before it?

Comment (nrc) [reply src/wasm-lib/kcl/src/parser/parser_impl.rs:947]: I'm pretty sure it works today without it, e.g., in the example here: https://zoo.dev/docs/kcl/reduce

I guess we should require something, or else there will be a back-compat hazard with parsing and `(...) {...}` is a little subtle for a function declaration.

Requiring `fn` seems like a good idea, but let me have a think

Comment (jtran) [reply src/wasm-lib/kcl/src/parser/parser_impl.rs:947]: It works today without fn, but we require =>. I honestly wouldn't worry about breaking code in this case. Anyone using anonymous functions isn't going to have a problem upgrading.

Comment (nrc) [reply src/wasm-lib/kcl/src/parser/parser_impl.rs:962]: Implemented in the latest commit

**Review by jtran — APPROVED** [review]
Awesome! Thanks for taking a second pass on this. 

Docs for map look like they don't use `fn`. Might need a regen or something.

**Comment by nrc** [comment]
> Docs for map look like they don't use fn. Might need a regen or something.

This doesn't require `fn` if you're using the old `=>` syntax, once you drop the arrow then you must use the `fn`.

**Comment by nrc** [comment]
> > Docs for map look like they don't use fn. Might need a regen or something.
> 
> This doesn't require `fn` if you're using the old `=>` syntax, once you drop the arrow then you must use the `fn`.

Oh sorry, you meant the generated docs, I think

**Comment by jtran** [comment]
image

CI seems to have passed, which I didn't expect.

File: docs/kcl/map.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ circles = map([1..3], drawCircle)
 ```js
 r = 10 // radius
 // Call `map`, using an anonymous function instead of a named one.
-circles = map([1..3], (id) => {
+circles = map([1..3], (id) {
```
Comment (jtran) [inline docs/kcl/map.md:48]: How is this okay?

Comment (nrc) [reply docs/kcl/map.md:48]: Seems like the docs didn't get updated - there is a `fn` in the doc comment

**Pull request merged**